src/lib/pickles/common.ml

```diff
@@ -235,23 +235,64 @@ let tick_public_input_of_statement ~max_proofs_verified ~feature_flags
     (Backend.Tick.Field.Vector.length input)
     ~f:(Backend.Tick.Field.Vector.get input)
 
+let maybe_scale ~scale g x l =
+  match x with None -> l | Some fp -> scale g fp :: l
+
 let ft_comm ~add:( + ) ~scale ~endoscale ~negate
     ~verification_key:(m : _ Plonk_verification_key_evals.t) ~alpha
     ~(plonk : _ Types.Wrap.Proof_state.Deferred_values.Plonk.In_circuit.t)
-    ~t_comm =
-  let ( * ) x g = scale g x in
+    ~t_comm ~options =
+  let _ = options in
+
+  (* ignore flags for now *)
   let _, [ sigma_comm_last ] =
     Vector.split m.sigma_comm (snd (Plonk_types.Permuts_minus_1.add Nat.N1.n))
   in
+  let ( * ) x g = scale g x in
+  (* [cond_mul] conditionally scale elements based on the relevant feature
+     flags, and their values.
+
+     Whenever anything is "misaligned" between the 3 arguments, this function
+     returns [None]. *)
+  let cond_mul flag sc comm =
```

Comment:
IIUC, calling this `maybe_mul` (or the `maybe_scale` above `cond_scale`) could be more consistant.